src/openforms/prefill/contrib/family_members/filters.py

```diff
@@ -0,0 +1,133 @@
+from datetime import datetime
+from typing import Literal
+
+from openforms.contrib.haal_centraal.clients import get_brp_client
+from openforms.contrib.haal_centraal.clients.brp import NaturalPersonDetails
+from openforms.contrib.haal_centraal.constants import BRPVersions
+from openforms.utils.date import get_today
+
+from .constants import HC_DECEASED_ATTRIBUTES
+
+
+def exclude_deceased_for_haal_centraal(
+    results: list[NaturalPersonDetails],
+    hc_version: Literal[BRPVersions.v13, BRPVersions.v20],
+) -> list[NaturalPersonDetails]:
+    bsn_numbers = [member["bsn"] for member in results]
+
+    with get_brp_client() as client:
+        # v2 of HaalCentraal supports request for multiple bsn numbers
+        if hc_version == BRPVersions.v20:
+            members = client.find_person(
+                bsn_numbers, attributes=HC_DECEASED_ATTRIBUTES[hc_version]
+            )
+        # v1 of HaalCentraal requires separate request for each bsn
+        else:
+            members = []
+            for bsn in bsn_numbers:
+                member = client.find_person(
+                    bsn, attributes=HC_DECEASED_ATTRIBUTES[hc_version]
+                )
+                if member:
+                    members.append(member)
+
+        if not members:
+            return results
+
+        deceased_bsns = [
+            member["burgerservicenummer"]
+            for member in members
+            if isinstance(member, dict) and "overlijden" in member
+        ]
+        eligible_members = [
+            result for result in results if result["bsn"] not in deceased_bsns
+        ]
+        return eligible_members
+
+
+def is_eligible_based_on_age(
+    provided_age: int | None = None,
+    date_of_birth: str | None = None,
+    date_of_birth_precision: Literal["date", "year_month", "year"] | None = None,
+    min_age: int | None = None,
+    max_age: int | None = None,
+) -> bool:
+    # Haal Centraal v1
+    if provided_age:
+        age = provided_age
+
+    # Haal Centraal v2 and StUF-BG
+    elif date_of_birth:
+        date_of_birth_obj = None
+        today_obj = datetime.strptime(get_today(), "%Y-%m-%d")
```

Comment:
I'd use `datetime_in_amsterdam(timezone.now()).date()` instead to avoid the extra roundtrip of parsing. Perhaps you can add `get_today_date()` in the utils that does that?

Reply:
I thought `get_today` does that already, perhaps I am missing something here..
https://github.com/open-formulieren/open-forms/blob/master/src/openforms/utils/date.py#L84